docs/proposals/naming-guidelines.md

```diff
@@ -0,0 +1,51 @@
+# Naming Guidelines
+
+One of the [two hard things](https://martinfowler.com/bliki/TwoHardThings.html) in Compute Science is naming things. It is proposed that Akri adopt naming guidelines to make developers' lives easier by providing consistency and reduce naming complexity.
+
+Akri existed before naming guidelines were documented and may not employ the guidelines summarized here. However, it is hoped that developers will, at least, consider these guidelines when extending Akri.
+
+## General Principles
+
++ Akri uses English
++ Akri is written principally in Rust and Rust [Naming](https://rust-lang.github.io/api-guidelines/naming.html) conventions are used
++ Types need not be included in names unless ambiguity would result
++ Shorter, simpler names are preferred
+
+## Akri Discovery Handlers
+
+Various Discovery Handlers have been developed: `debug_echo`, `http`, `onvif`, `opcua`, `udev`
+
+Guidance:
+
++ `snake_case` names
++ (widely understood) initializations|acronyms are preferred
+
+## Akri Brokers
+
+Various Brokers have been developed: `http-broker`, `onvif-video-broker`, `opcua-monitoring-broker`
```

Comment:
Maybe add the `udev-video-broker` for completeness.

Reply:
Fixed: https://github.com/deislabs/akri/pull/204/commits/40efa54ad1a1a9d3f0d9fc5a0648abf01475e691